telemetry/influxdb/src/main/scala/io/buoyant/telemetry/influxdb/InfluxDbTelemeter.scala

```diff
@@ -0,0 +1,123 @@
+    } else {
+      ""
+    }
+
+  private[this] def labelExists(labels: Seq[(String, String)], name: String) =
+    labels.exists(_._1 == name)
+
+  private[this] def writeMetrics(
+    tree: MetricsTree,
+    sb: StringBuilder,
+    prefix0: Seq[String],
+    tags0: Seq[(String, String)]
+  ): Unit = {
+    // TODO: share with prometheus telemeter?
+    // Re-write elements out of the prefix into labels
+    val (prefix1, tags1) = prefix0 match {
+      case Seq("rt", router) if !labelExists(tags0, "rt") =>
+        (Seq("rt"), tags0 :+ ("rt" -> router))
+      case Seq("rt", "dst", "path", path) if !labelExists(tags0, "dst_path") =>
+        (Seq("rt", "dst_path"), tags0 :+ ("dst_path" -> path))
+      case Seq("rt", "dst", "id", id) if !labelExists(tags0, "dst_id") =>
+        (Seq("rt", "dst_id"), tags0 :+ ("dst_id" -> id))
+      case Seq("rt", "dst_id", "path", path) if !labelExists(tags0, "dst_path") =>
+        (Seq("rt", "dst_id", "dst_path"), tags0 :+ ("dst_path" -> path))
+      case Seq("rt", "srv", srv) if !labelExists(tags0, "srv") =>
+        (Seq("rt", "srv"), tags0 :+ ("srv" -> srv))
+      case _ => (prefix0, tags0)
+    }
+
+    // gather all metrics directly under a common parent
+    val fields = {
+      tree.children.toSeq map {
+        case (name, child) =>
+          // write metrics deeper in the tree (side effect)
+          writeMetrics(child, sb, prefix1 :+ name, tags1)
+
+          child.metric match {
+            case c: Metric.Counter => Seq(Some((name, c.get.toString)))
+            case g: Metric.Gauge => Seq(Some((name, g.get.toString)))
+            case s: Metric.Stat =>
+              val summary = s.snapshottedSummary
+              if (summary != null) {
+                Seq(
+                  Some((name + "_count", summary.count.toString)),
+                  Some((name + "_sum", summary.sum.toString)),
+                  Some((name + "_avg", summary.avg.toString)),
+                  Some((name + "_min", summary.min.toString)),
+                  Some((name + "_max", summary.max.toString)),
+                  Some((name + "_p50", summary.p50.toString)),
+                  Some((name + "_p90", summary.p90.toString)),
+                  Some((name + "_p95", summary.p95.toString)),
+                  Some((name + "_p99", summary.p99.toString)),
+                  Some((name + "_p999", summary.p9990.toString)),
+                  Some((name + "_p9999", summary.p9999.toString))
+                )
+              } else {
+                Seq(None)
+              }
+            case _ => Seq(None)
+          }
+      }
+    }.flatten.flatten
+
+    // write sibling metrics as fields in a single measurement
+    if (fields.nonEmpty) {
+      sb.append(escapeKey(prefix1.mkString(":")))
+      if (tags1.nonEmpty) {
+        sb.append(",")
+        sb.append(formatLabels(tags1))
+      }
+      sb.append(" ")
+      sb.append(formatLabels(fields))
+      sb.append("\n")
+    }
+
+    return
```

Comment:
Oh.  The problem is that the method return type is `Unit` but `append` returns a stringbuilder.  You can do `val _ = sb.append("\n")` to explicitly discard the returned stringbuilder.